src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -113,121 +156,299 @@ export const MultiSelectDialog = chakra(
+              marginTop="s"
+            >
+              {viewAllLabel}
+            </Text>
+          );
+        }
+      };
+
+      const itemsNotFound = () => {
+        if (itemsList.length === 0) {
+          return (
+            <Text id="text-btn" marginTop="xs" isItalic="true">
+              No options found
+            </Text>
+          );
+        }
+      };
+
+      // We need this for our "fake" button inside the main menu button.
+      function onKeyPress(e) {
+        const enterOrSpace =
+          e.key === "Enter" ||
+          e.key === " " ||
+          e.key === "Spacebar" ||
+          e.which === 13 ||
+          e.which === 32;
+
+        if (enterOrSpace) {
+          e.preventDefault();
+          onClear();
+          internalRef?.current.focus();
+        }
+      }
+      // Manage focus upon closing the MultiSelect
+      // React.useEffect(() => {
+      //   setPrevIsOpen(isDropDownOpen);
+      //   if (isDropDownOpen) {
+      //   // internalRef.current?.click();
+      //   }
+      // }, [isDropDownOpen]);
+
+      const viewAllItems = () => {
+        setViewAllLabel(viewAllLabel === "View all" ? "View less" : "View all");
+        setListHeight(listHeight === "260px" ? "" : "260px");
+        setListItemsCount(
+          listItemsCount === defaultItemsVisible
+            ? items.length
+            : defaultItemsVisible
+        );
+        if (listItemsCount === defaultItemsVisible) {
+          setItemsList(items);
+        } else {
+          displayDefaultItems();
+        }
+      };
+
+      const buttonTextLabel = () => {
+        return (
+          <Text __css={styles.accordianLabel} title={buttonText}>
+            {truncateText(buttonText as string, 20)}
+          </Text>
+        );
+      };
+
+      return (
+        <Box
+          id={id}
+          ref={mergedRefs}
+          __css={styles}
+          {...rest}
+          onKeyPress={onKeyPress}
+        >
+          <MultiSelectMenuButton
+            id={`ms-${id}-menu-button`}
+            multiSelectId={id}
+            multiSelectLabelText={buttonText}
+            isOpen={isOpen}
+            selectedItems={selectedItems}
+            onClear={onClear}
+          />
+          <Accordion
+            sx={{
+              _first: {
+                "> div": {
+                  button: styles.menuButton,
+                  "> div": {
+                    button: {minWidth: "0px"},
+                  }
+                }
+              },
+              "> div": {
+                _first: {
+                  "> div": {
+                    _first: {
+                      minWidth: "200px",
+                      maxWidth: "300px",
+                      borderTopWidth: "0.5px",
+                    },
+                  },
+                },
+              },
+            }}
+            accordionData={[
+              {
+                accordionType: "default",
+                label: buttonTextLabel(),
```

Comment:
We expect `label` to be a string. So the `buttonTextLabel` function can return just a string.